invenio_app_ils/ui/src/invenio_app_ils/forms/components/CountryField.js

```diff
@@ -0,0 +1,329 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { VocabularyField } from '../core';
+import { Dropdown, Flag } from 'semantic-ui-react';
+import cx from 'classnames';
+import get from 'lodash/get';
+
+export class FlagDropdown extends Dropdown {
+  renderText = () => {
+    const { multiple, placeholder, search, text } = this.props;
+    const { searchQuery, value, open } = this.state;
+    const hasValue = this.hasValue();
+
+    const classes = cx(
+      placeholder && !hasValue && 'default',
+      'text',
+      search && searchQuery && 'filtered'
+    );
+    let _text = placeholder;
+    let _flag = null;
+
+    if (text) {
+      _text = text;
+    } else if (open && !multiple) {
+      _text = get(this.getSelectedItem(), 'text');
+    } else if (hasValue) {
+      const item = this.getItemByValue(value);
+      _text = get(item, 'text');
+      if (item && item.flag) {
+        _flag = <Flag name={item.flag} />;
+      }
+    }
+
+    // Render the flag along with the text
+    return (
+      <div className={classes} role="alert" aria-live="polite" aria-atomic>
+        {_flag}
+        {_text}
+      </div>
+    );
+  };
+}
+
+export const FLAG_NAMES = [
```

Comment:
is there any chance we can take them from a vocabulary?

Reply:
The problem is that semantic UI throws an exception if it gets an incorrect flag name so I need to compare with the list of valid names. I just saw that semantic exports the list of valid flag names so I'll use that instead.

Reply:
If you don't like it, we can get rid of the flags! :)